index.html

```diff
@@ -179,44 +194,34 @@
                         if (enemy.y < Crafty('PlayerCharacter').y) {
                             enemy.velocity().y = 100
                         } 
-                        if (-30 < enemy.x - Crafty('PlayerCharacter').x < 30 || -30 < enemy.y - Crafty('PlayerCharacter').y < 30)
-                            var villain = this;
-                            villain.bind('Moved', function(movement) {
-                                if (movement.axis == "x") {
-                                    if (villain.x - movement.oldValue < 0) {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.LEFT;
-                                    } else {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.RIGHT;
-                                    }
-                                } else {
-                                    if (villain.y - movement.oldValue < 0) {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.UP;
-                                    } else {
-                                        villain._attackDirection = ATTACK_DIRECTIONS.DOWN;
-                                    }
-                                };
-                            });
-                            enemy.velocity() = 0;
+
+                        var x_distance = Math.abs(enemy.x - Crafty('PlayerCharacter').x);
```

Comment:
your original code seemed right but something was busted so I just did it this way